### DISCUSSION TITLE
RexUI InputText: Trying to adjust InputText object when resizing the window/screen

### DISCUSSION THREAD

**Discussion by jcoppage** [rexrainbow/phaser3-rex-notes#220] [Q&A] [answered]
For the life of me I can not seem to resize and re-position my InputText Object at the same time when my resize event is called.  When I call .setPosition() it works great. But when I call .resize(), it changes it to the correct size, but it changes the InputText (x,y) position for some reason.  The way it gets repositioned, I'm guessing by calling the resize() method it changes its (x,y) anchor origin point????  I've even tried: this.inputText.node.width = width; this.inputText.updateSize(); and still get the same results. I've been working on this for a while and its getting very frustrating.  The only other option I can think of is just destroying and re-creating the TextInput object every time my window is resized, but I feel like that would take up a lot of resources. 
<img width="453" alt="Screen Shot 2021-12-27 at 10 26 20 PM" src="https://user-images.githubusercontent.com/70357667/147535175-908bf86c-334d-41a2-a650-651bafa8eae1.png">

**Comment by rexrainbow** [comment]
Could you provide a simplest, run-able test code for this case?

**Reply by jcoppage** [reply]
Reproduced the scenario and it worked correctly.  Went back to my code and found the problem: had the alphas set to 0 (because of a previous situation) while I was doing the resizing.  Set alphas back to 1 before the resizing and now everything  works perfectly now.  My apologies for the trouble.  Thanks.

**Comment by jcoppage** [comment, marked as answer]
SOLUTION: Had the alphas set to 0 while I was doing the resizing.  Changed to make sure alphas were set back to 1 before the resizing calculations and it resizes/re-positions perfectly now. EDIT: Still somewhat acting up: I need to be able to resize/re-position the TextInput object while its invisible on the screen (when the player is not using it). Not working well switching from setAlpha(0) to setAlpha(1).  
SOLUTION #2: instead of setting the alpha, type: this.inputText.node.style.visibility = “hidden"; // or “visible”
With this you can resize/re-position the InputText object all you want, it doesn't matter if it's "hidden" or "visible"
